hw/ip/sysrst_ctrl/data/sysrst_ctrl_testplan.hjson

```diff
@@ -0,0 +1,214 @@
+
+    {
+      name: keyboard_input_triggered_interrupt
+      desc: '''
+            Verify the keyboard and input triggered interrupt feature by detecting the edge transitions on input pins.
+
+            * Set the input signals and edge transition by configuring KEY_INTR_CTL register.                                                                                             
+            * Set the debounce timer value via KEY_INTR_DEBOUNCE_CTL register.                                                                                                        
+            * Read the KEY_INTR_STATUS register to check if the interrupt caused and clear the interrupt.
+            '''
+      milestone: V2
+      tests: ["sysrst_ctrl_edge_detect"]
+    }
+
+    {
+      name: pin_output_keyboard_inversion_control
+      desc: '''
+            Verify the keyboard inversion feature by override logic.
+
+            * Select the output signals to override via PIN_OUT_CTL register. 
+            * Allow the output signals to override the value via PIN_ALLOWED_CTL register.                                                                                                                                                           * Set the override value to the output signal via PIN_OUT_VALUE register.
+            '''
+      milestone: V2
+      tests: ["sysrst_ctrl_override_test"]
+    }
+
+    {
+      name: pin_input_value_accessibility
+      desc: '''
+            Verify the pin input value accessibilty.
+
+            * Trigger the key[0,1,2], ac_present_in, pwrb_in, ec_rst_in_l input pins with random values. 
+            * Read the PIN_IN_VALUE register and check if the read value is same as input value.                                                                                                                                         
+            '''
+      milestone: V2
+      tests: ["sysrst_ctrl_pin_access_test"]
+    }
+
+    {
+      name: ec_power_on_reset
+      desc: '''
+            Verify the EC and power on reset.
+
+            * Enable the override value by writing to PIN_ALLOWED_CTL.EC_RST_L_1 and PIN_OUT_CTL.EC_RST_L register. 
+            * Make sure ec_rst_out_l is asserted even after opentitan reset is released.
+            * Set PIN_OUT_CTL.EC_RST_L to 0 to release the ec_rst_out_l reset.
+            '''
+      milestone: V2
+      tests: ["sysrst_ctrl_ec_pwr_on_rst"]
+    }
+
+    {
+      name: flash_write_protect_output
+      desc: '''
+            Verify the flash write protect.
+
+            * Make sure flash_wp_out_l signal is asserted low by reading PIN_OUT_CTL.flash_wp_l.
+            * Enable the override function by setting PIN_OUT_CTL.FLASH_WP_L to value 1. 
+            * Check if flash_wp_out_l is released. 
+            '''
+      milestone: V2
+      tests: ["sysrst_ctrl_flash_wr_prot_out"]
+    }
+
+    {
+      name: ultra_low_power_test
+      desc: '''
+            Verify the ultra low power feature. 
```

Comment:
It would be good to mention here that the bus clock should be disabled as part of this this test in order to check whether the detection logic actually works in sleep mode.

I.e., the `sysrst_ctrl` should first assert `wkup_req_o` before `clk_i` is turned on again in order to read out and check the `WKUP_STATUS` and `ULP_STATUS` registers.

Reply:
Updated.